### ISSUE TITLE
RuntimeError: Expected tensor for argument #1 'indices' to have scalar type Long; but got torch.IntTensor instead 

### ISSUE THREAD

**Issue by JumpyPizza** [UKPLab/sentence-transformers#13]
Everything was just fine before I reinstalled my anaconda to the latest version(2019.07)(or it could be the update to 0.2.2?), but now when I run the code it gives this Runtime Error:

 File "D:\anaconda\lib\site-packages\sentence_transformers\SentenceTransformer.py", line 136, in encode
    embeddings = self.forward(features)
  File "D:\anaconda\lib\site-packages\torch\nn\modules\container.py", line 92, in forward
    input = module(input)
  File "D:\anaconda\lib\site-packages\torch\nn\modules\module.py", line 489, in __call__
    result = self.forward(*input, **kwargs)
  File "D:\anaconda\lib\site-packages\sentence_transformers\models\BERT.py", line 27, in forward
    output_tokens = self.bert(input_ids=features['input_ids'], token_type_ids=features['token_type_ids'], attention_mask=features['input_mask'])[0]
  File "D:\anaconda\lib\site-packages\torch\nn\modules\module.py", line 489, in __call__
    result = self.forward(*input, **kwargs)
  File "D:\anaconda\lib\site-packages\pytorch_transformers\modeling_bert.py", line 712, in forward
    embedding_output = self.embeddings(input_ids, position_ids=position_ids, token_type_ids=token_type_ids)
  File "D:\anaconda\lib\site-packages\torch\nn\modules\module.py", line 489, in __call__
    result = self.forward(*input, **kwargs)
  File "D:\anaconda\lib\site-packages\pytorch_transformers\modeling_bert.py", line 264, in forward
    words_embeddings = self.word_embeddings(input_ids)
  File "D:\anaconda\lib\site-packages\torch\nn\modules\module.py", line 489, in __call__
    result = self.forward(*input, **kwargs)
  File "D:\anaconda\lib\site-packages\torch\nn\modules\sparse.py", line 118, in forward
    self.norm_type, self.scale_grad_by_freq, self.sparse)
  File "D:\anaconda\lib\site-packages\torch\nn\functional.py", line 1454, in embedding
    return torch.embedding(weight, input, padding_idx, scale_grad_by_freq, sparse)
RuntimeError: Expected tensor for argument #1 'indices' to have scalar type Long; but got torch.IntTensor instead (while checking arguments for embedding)

I've reinstalled pytorch and pytorch-transformers, but it doesn't help. Here is my simple code:

model = SentenceTransformer("./model/")
sentences = ['This framework generates embeddings for each input sentence',
                 'Sentences are passed as a list of string.',
                 'The quick brown fox jumps over the lazy dog.']
sentence_embeddings = model.encode(sentences)

where model is bert-base-nli-mean-tokens. 

Any help would be appreciated! Thanks a lot!

**Comment by nreimers** [comment]
Could you check which version you have?

You can get it with:
```
import sentence_transfomers
print(sentence_transfomers.__version__
```

Can you also try this code:
```
model = SentenceTransformer("bert-base-nli-mean-tokens")
sentences = ['This framework generates embeddings for each input sentence',
'Sentences are passed as a list of string.',
'The quick brown fox jumps over the lazy dog.']
sentence_embeddings = model.encode(sentences)
```

I sadly haven't tested it yet on Windows with Anaconda.

**Comment by JumpyPizza** [comment]
Thanks very much for your quick response!!
My version is 0.2.2, and I've tested with model = SentenceTransformer("bert-base-nli-mean-tokens"), but it still has the same error.. But strange thing is it worked just fine before I reinstalled anaconda and sentence_transformers. I was guessing it had something to do with torch's version, but after I reinstalled pytorch it still doesn't work.
My torch's version is 1.0.1, pytorch-transformers is 1.1.0 and python is 3.7.3.

**Comment by JumpyPizza** [comment]
I've added a .long() to cast sentence features to long here after line 
https://github.com/UKPLab/sentence-transformers/blob/2a0de8846b3e7b50644463e0d17f106b1b0672c3/sentence_transformers/SentenceTransformer.py#L133
Then somehow the problem is solved. But still I can't figure out how it worked?

**Comment by nreimers** [comment]
@JumpyPizza 
I can reproduce your error on Anaconda on Windows (with Ubuntu and conda, everything works). I think that torch.from_numpy() gets on Windows a wrong type: An int type and not a long type.

I will try to publish a fix as soon as possible.

**Comment by nreimers** [comment]
@JumpyPizza I fixed the issue in the latest version (0.2.3). Just update with "pip install -U sentence-transformers" and it should work on Windows. Tested it with Windows 10, Anaconda and Python 3.6.

If the issue still exists please let me know.

**Comment by JumpyPizza** [comment]
@nreimers Great! Thanks a lot for your excellent work!
